lib/foreman_maintain/cli/upgrade_command.rb

```diff
@@ -70,16 +31,30 @@ def execute
       end
 
       subcommand 'run', 'Run full upgrade to a specified version' do
-        target_version_option
         interactive_option
         disable_self_upgrade_option
 
         def execute
           ForemanMaintain.validate_downstream_packages
           ForemanMaintain.perform_self_upgrade if allow_self_upgrade?
-          upgrade_runner.run
-          upgrade_runner.save
-          exit upgrade_runner.exit_code
+
+          if upgrade_runner.available?
```

Comment:
Should the `check` command get the same treatment?
Otherwise you get:
```console
[root@sat ~]# foreman-maintain upgrade check
Checking for new version of satellite-maintain...
Nothing to update, can't find new version of satellite-maintain.
[root@sat ~]# echo $?
0
```

Reply:
Done. with a little re-factoring to clean up the code for re-use.